pkg/providers/v1/providers.go

```diff
@@ -133,3 +134,12 @@ func ParseAndValidate(rawConfig json.RawMessage, to any) error {
 
 	return nil
 }
+
+// As is a type-cast function for Providers
+func As[T Provider](provider Provider) (T, error) {
```

Comment:
is this used anywhere?

Reply:
It will be used in the next PR. It will be used as a helper function for casting the Provider to a specific trait